packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -619,7 +623,7 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
       formatters.add(LengthLimitingTextInputFormatter(widget.maxLength));
     }
     final CupertinoThemeData themeData = CupertinoTheme.of(context);
-    final TextStyle textStyle = widget.style ?? themeData.textTheme.textStyle;
+    final TextStyle textStyle = themeData.textTheme.textStyle.merge(widget.style);
```

Comment:
This is technically a breaking change.  Can you elaborate on why this was changed?  Also, do you plan to announce this breaking change when merged?

Reply:
I think I should talk to someone more familiar with Cupertino about the exact desired behavior here. This is the behavior in material, and the way strut inherits the textStyle's properties is dependent on the textstyle being complete.

Reply:
@xster

Reply:
Talked with xster and we have agreed the change here is reasonable. I have sent a breaking change announce out and added the API break tag. The change here should be minor and can be bypassed with the `inherit: false` parameter in TextStyle